internal/provider/data_source_dirhash_sha256.go

```diff
@@ -0,0 +1,60 @@
+package provider
+
+import (
+	"context"
+
+	"github.com/Think-iT-Labs/dirhash/lib"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func dataSourceDirhashSha256() *schema.Resource {
+	return &schema.Resource{
+		Description: "Calculate SHA256 checksum of a given directory",
+
+		ReadContext: dataSourceDirhashSha256Read,
+
+		Schema: map[string]*schema.Schema{
+			"directory": {
+				Type:        schema.TypeString,
+				Description: "Directory to hash.",
+				Required:    true,
+			},
+			"ignore": {
+				Type:        schema.TypeList,
+				Description: "List of ignored glob patterns.",
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Optional:    true,
+			},
+			"checksum": {
+				Type:        schema.TypeString,
+				Description: "Directory SHA 256 Checksum",
+				Computed:    true,
+			},
+		},
+	}
+}
+
+func dataSourceDirhashSha256Read(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
+	var dir = d.Get("directory").(string)
+	d.SetId(dir)
+	var ignore = []string{}
+	if v, ok := d.GetOk("ignore"); ok {
+		ignore = stringListToStringSlice(v.([]interface{}))
+	}
+	var checksum = lib.DirHash(dir, ignore)
```

Comment:
I suggest we check for errors here as an attempt to hash a directory can, at least in principle,fail. (no read access, directory does not exist, invalid path ...). 

I even took a look at the lib code and seemingly it also exits without returning an error. won't that cause the provider to fail with no output to the user?

Reply:
I would go and continue without this, but I will create an issue for it in the library/cli repository